0099-Replacing bash as the default shell.md

```diff
@@ -0,0 +1,121 @@
+---
+feature: oil-shell
+start-date: 2021-08-11
+author: Raphael Megzari (happysalada)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues:
+---
+
+# Summary
+
+[summary]: #summary
+
+Use oilshell as the default shell for nixos. Oilshell aims to be compatible with bash while trying to bring real programming language features (like hashmap or dicts) to the shell.
+
+## Motivation
+
+[motivation]: #motivation
+
+Two primary motivations
+
+- Removing the footguns in bash that even experienced programmers find painful
+- Making shell scripts more powerfull to remove the need to bring additional languages for scripting.
+
+## Detailed design
+
+[design]: #detailed-design
+
+The [oil shell](https://www.oilshell.org/) has two parts osh and oil. Osh is compatible with bash and posix and it's goal is to run existing shell scripts. The oil language is a brand new incompatible language. The idea is to fix more than four decades of accumulated warts in the Unix shell. Many Unix users are angry that shell is so difficult, and Oil aims to fix that. Those definitions were taken verbatim from [reference](https://www.oilshell.org/blog/2021/01/why-a-new-shell.html)
```

Comment:
How do osh and oil interoperate?

Reply:
You have two binaries, and depending on the one you use, you have different features. osh can run bash script almost as is. In order for oil to run bash scripts, there needs to run scripts, those need to be reworked to tolerate the `strict err_exit`. Running a script with oil enables you to use additional features of the oil language. Where scripts run with osh, only have access to bash shell features. Let me know if this doesn't answer your question.

Reply:
So you can access functions defined in an Oil script I assume?

Reply:
In the same way you can access bash function defined in a script.
oil introduce procs which are functions that don't mess with their outer scope and take parameters. I don't know if the same interoperability would apply to those. The way to call a proc is to pass it arguments. I think those would be fine too but this is something that has to be confirmed.

Reply:
To clarify, "oilshel"l is a standardization of the POSIX shell protocol, and "Oil" (the programming language) is oilshell augmented with Oil language features, correct?

Reply:
osh is a bit more as it is meant to run bash things which have some non POSIX behaviours in it. But yes, oil is osh augmented with new language features.